lib/memory_profiler/reporter.rb

```diff
@@ -100,12 +106,19 @@ def object_list(generation)
           class_name = helper.lookup_class_name(klass)
           gem        = helper.guess_gem(file)
 
-          string     = '' << obj  if klass == String
-
           memsize = ObjectSpace.memsize_of(obj) + rvalue_size_adjustment
           # compensate for API bug
           memsize = rvalue_size if memsize > 100_000_000_000
-          result[obj.__id__] = MemoryProfiler::Stat.new(class_name, gem, file, location, memsize, string)
+
+          if klass == String
+            if mem_lite
+              string = helper.string_summary(obj)
+              md5    = helper.lookup_string_digest(obj)
+            else
+              string = '' << obj  if klass == String
```

Comment:
Just noticed that the `if klass == String` is redundant, and is a relic from some copy-pasta.  Oops, will fix.